### PR TITLE
[676] Only show edit button when signer is an owner

### LINKED ISSUE #676: project details: remove edit button and form access when i'm not the project owner
**Is your feature request related to a problem? Please describe.**
When I open a project i don't own (eg. @michellema1208 's project: https://staging.grantshub.gitcoin.co/?debug=1/#/chains/5/registry/0x832c5391dc7931312CbdBc1046669c9c3A4A28d5/projects/11)
I see the edit button and i'm able to access the edit form, only when i try to save and publish it fails.

**Describe the solution you'd like**
Only project owner should see the edit button and should be able to access the edit form.

Tasks: 

- [ ] Remove edit button from Project Details page, if being viewed by a non-project owner 
- [ ] Remove access to edit page for a project if being accessed by a non-project owner

### PR DESCRIPTION
**Pull request by boudra**
Resolves #676

### CODE REVIEW

**Review by codenamejason — DISMISSED** [review]
Should we add a test case for this? I had approved but switched project to another on another chain and the edit button did not display and I am the owner.

**Comment by boudra** [comment]
hmm the signer address changes when you switch networks on your wallet right? not sure how we can identify whether the user is the owner in that scenario

**Comment by codenamejason** [comment]
The signer should carry over to whatever chain your wallet has selected. I think with Rainbow/wagmi they just sign in once and can traverse chain to chain.

File: client/src/reducers/projects.ts [hunk 1]
```diff
@@ -31,22 +32,26 @@ export type Application = {
   chainId: number;
 };
 
+export type ProjectOwners = { [key: string]: string[] };
```
Comment (gravityblast) [inline client/src/reducers/projects.ts]:
```suggestion
export type ProjectOwners = { [projectID: string]: string[] };
```

**Review by gravityblast — APPROVED** [review]

**Review by michellema1208 — APPROVED** [review]

**Pull request merged**